massdriver-application/main.tf

```diff
@@ -29,11 +29,10 @@ locals {
 
   # App Serivce / Function App will auto-inject the secret, but still need these env-vars added.
   # AKS auto-injects everything (we add the Client ID and Tenant ID as annotations).
-  # I don't think having these added to all runtimes will hurt Workload Identity or other auth solutions.
-  azure_envs = {
-    AZURE_CLIENT_ID = mdxc_application_identity.main.azure_application_identity.client_id
-    AZURE_TENANT_ID = mdxc_application_identity.main.azure_application_identity.tenant_id
-  }
+  azure_envs = local.is_azure && mdxc_application_identity.main.azure_application_identity != null ? {
+    AZURE_CLIENT_ID = try(mdxc_application_identity.main.azure_application_identity.client_id, "")
```

Comment:
We should add a `envs` field to mdxc_application_identity that returns a map[string]string of envs that the cloud needs so we don't have to have this cloud specific code here. Wed just be able to merge that map into envs... which would be `{}` by default.

Reply:
Another option is burying this code in the `massdriver-application-azure-app-service` and `massdriver-application-azure-function-app` modules. This code isn't needed for **Azure**, its only needed for those 2 runtimes. This way we encode the runtime dependencies into the runtime modules instead of having it at this level.

Reply:
I think this commit: https://github.com/massdriver-cloud/terraform-modules/pull/125/commits/59e9a81be1038630dabd5149c26733b6fe7df4dc
Address this the best way.